client/python/tests/integration/test_no_auth.py

```diff
@@ -9,86 +9,95 @@
+    if os.environ.get("ARMADA_SERVER"):
+        server_name = os.environ.get("ARMADA_SERVER")
 
-def test_submit_job():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
+    if os.environ.get("ARMADA_PORT"):
+        server_port = os.environ.get("ARMADA_PORT")
+
+    return ArmadaClient(
+        channel=grpc.insecure_channel(target=f"{server_name}:{server_port}")
     )
-    # Jobs can must be finished before deleting queue
-    no_auth_client.cancel_jobs(queue=queue_name, job_set_id=job_set_name)
-    no_auth_client.delete_queue(name=queue_name)
-    time.sleep(1)
 
 
-def test_submit_job_and_cancel_by_id():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    jobs = no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
-    )
-    # Jobs can must be finished before deleting queue
-    no_auth_client.cancel_jobs(job_id=jobs.job_response_items[0].job_id)
-    no_auth_client.delete_queue(name=queue_name)
+no_auth_client = client()
 
 
-def test_submit_job_and_cancel_by_queue_job_id():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
-    )
-    # Jobs can must be finished before deleting queue
-    no_auth_client.cancel_jobs(queue=queue_name, job_set_id=job_set_name)
-    no_auth_client.delete_queue(name=queue_name)
+def sleep(sleep_time):
+    time.sleep(sleep_time)
 
 
-def test_get_queue():
-    queue_name = f"queue-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
+queue_name = f"queue-{uuid.uuid1()}"
+
+
+@pytest.fixture(scope="session", autouse=True)
+def queue():
+
+    no_auth_client.create_queue(name=queue_name, priority_factor=1)
+    sleep(3)
+
 
+def test_get_queue():
     queue = no_auth_client.get_queue(name=queue_name)
     assert queue.name == queue_name
-    no_auth_client.delete_queue(name=queue_name)
 
 
 def test_get_queue_info():
-    queue_name = f"queue-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-
     queue = no_auth_client.get_queue_info(name=queue_name)
     assert queue.name == queue_name
     assert not queue.active_job_sets
-    no_auth_client.delete_queue(name=queue_name)
+
+
+def test_submit_job_and_cancel_by_id():
+    job_set_name = f"set-{uuid.uuid1()}"
+    jobs = no_auth_client.submit_jobs(
+        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
+    )
+    sleep(5)
```

Comment:
Is there anything we can do to avoid sleeping here? These will be run in sequence which means we're adding at least 20 seconds to test runtime. Any reason we can't poll to see when whatever we're waiting on (the job to actually be submitted?) to be complete?

Reply:
The problem is that we have to sleep before we start polling.  When you submit a job, it may take some time for events to pick it up.

I have to deal with this in my job service where a job polling fails until the first event.

Reply:
This has been handled  :))